pkg/ingester/ingester.go

```diff
@@ -77,15 +81,20 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	cfg.WALConfig.RegisterFlags(f)
 
 	f.IntVar(&cfg.MaxTransferRetries, "ingester.max-transfer-retries", 10, "Number of times to try and transfer chunks before falling back to flushing. Negative value or zero disables hand-over.")
+
 	f.DurationVar(&cfg.FlushCheckPeriod, "ingester.flush-period", 1*time.Minute, "Period with which to attempt to flush chunks.")
 	f.DurationVar(&cfg.RetainPeriod, "ingester.retain-period", 5*time.Minute, "Period chunks will remain in memory after flushing.")
-	f.DurationVar(&cfg.FlushOpTimeout, "ingester.flush-op-timeout", 1*time.Minute, "Timeout for individual flush operations.")
 	f.DurationVar(&cfg.MaxChunkIdle, "ingester.max-chunk-idle", 5*time.Minute, "Maximum chunk idle time before flushing.")
 	f.DurationVar(&cfg.MaxStaleChunkIdle, "ingester.max-stale-chunk-idle", 2*time.Minute, "Maximum chunk idle time for chunks terminating in stale markers before flushing. 0 disables it and a stale series is not flushed until the max-chunk-idle timeout is reached.")
+	f.DurationVar(&cfg.FlushOpTimeout, "ingester.flush-op-timeout", 1*time.Minute, "Timeout for individual flush operations.")
 	f.DurationVar(&cfg.MaxChunkAge, "ingester.max-chunk-age", 12*time.Hour, "Maximum chunk age before flushing.")
 	f.DurationVar(&cfg.ChunkAgeJitter, "ingester.chunk-age-jitter", 0, "Range of time to subtract from -ingester.max-chunk-age to spread out flushes")
-	f.BoolVar(&cfg.SpreadFlushes, "ingester.spread-flushes", true, "If true, spread series flushes across the whole period of -ingester.max-chunk-age.")
 	f.IntVar(&cfg.ConcurrentFlushes, "ingester.concurrent-flushes", 50, "Number of concurrent goroutines flushing to dynamodb.")
+	f.BoolVar(&cfg.SpreadFlushes, "ingester.spread-flushes", true, "If true, spread series flushes across the whole period of -ingester.max-chunk-age.")
+
+	f.DurationVar(&cfg.MetadataPurgePeriod, "ingester.metadata-purge-period", 2*time.Minute, "Period at which to attempt purging metadata from memory.")
+	f.DurationVar(&cfg.MetadataRetainPeriod, "ingester.metadata-retain-period", 3*time.Minute, "Period at which metadata we have not seen will remain in memory before being deleted.")
```

Comment:
I had a similar thought process but decide that retain period was closer based on the discarded design. `MaxChunkIdle ` refers to the time before we commit the chunk (as in saving it) whereas `RetainPeriod` refers to the time before we remove it from memory. Given metadata (as you mentioned) is discarded by design this felt a bit closer to that. 

But more importantly, should we later decide to save metadata somewhere we would have to change the semantics of the flags at that point - I would like to avoid it if possible. 

tl;dr: In my head, "idle" refers to "time before saving" and "retain period" refers to "time before discarding" - that's why I chose the latter.